tests/blackbox_test.go

```diff
@@ -79,15 +91,35 @@ func TestMain(m *testing.M) {
 func TestIgnitionBlackBox(t *testing.T) {
 	for _, test := range register.Tests[register.PositiveTest] {
 		test := test
+		var uConfig IgnitionConfig
+		if test.ConfigVersion != "3.1.0" && test.ConfigVersion != "3.0.0" && test.Config != "" {
+			if err := json.Unmarshal([]byte(test.Config), &uConfig); err != nil {
+				t.Log(test.ConfigVersion)
+				t.Log(uConfig)
+				t.Error(fmt.Errorf("error parsing Ignition configuration: %v", err))
+			}
+		}
 		t.Run(test.Name, func(t *testing.T) {
+			defer func() {
+				for _, luks := range uConfig.Storage.Luks {
+					t.Log(luks.Name)
+					if err := removeLuksDevice(luks.Name); err != nil {
+						t.Error(fmt.Errorf("failed to remove existing LUKS device %s: %v", luks.Name, err))
+					}
+				}
+			}()
 			if killContext.Err() != nil || (testing.Short() && test.ConfigMinVersion != test.ConfigVersion) {
 				t.SkipNow()
 			}
 			if listSubtests {
 				fmt.Println(t.Name())
 				return
 			}
-			t.Parallel()
+			// Luks tests cannot be run in parallel as LUKS device names are unique.
+			if len(uConfig.Storage.Luks) == 0 {
+				t.Parallel()
+			}
```

Comment:
I think instead we should have something similar to `ReplaceAllUUIDVars()` here and not rely on tests using unique names.

Reply:
Oh; Ok, I can certainly do that; At one point I had created some sugar for that.